ambry-api/src/main/java/com.github.ambry/account/Container.java

```diff
@@ -0,0 +1,243 @@
+    }
+  }
+
+  /**
+   * Constructor that takes individual arguments. Cannot be null.
+   * @param id The id of the container. Cannot be null.
+   * @param name The name of the container. Cannot be null.
+   * @param status The status of the container. Cannot be null.
+   * @param description The description of the container. Can be null.
+   * @param isPrivate The privacy setting of the container. Cannot be null.
+   * @param parentAccountId The id of the parent {@link Account} of this container. Cannot be null.
+   */
+  Container(Short id, String name, ContainerStatus status, String description, Boolean isPrivate,
+      Short parentAccountId) {
+    this.id = id;
+    this.name = name;
+    this.status = status;
+    this.description = description;
+    this.isPrivate = isPrivate;
+    this.parentAccountId = parentAccountId;
+    checkRequiredFields();
+  }
+
+  /**
+   * Gets the id of the container.
+   * @return The id of the container.
+   */
+  public short getId() {
+    return id;
+  }
+
+  /**
+   * Gets the name of the container.
+   * @return The name of the container.
+   */
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * Gets the status of the container.
+   * @return The status of the container.
+   */
+  public ContainerStatus getStatus() {
+    return status;
+  }
+
+  /**
+   * Gets the description of the container.
+   * @return The description of the container.
+   */
+  public String getDescription() {
+    return description;
+  }
+
+  /**
+   * Gets the boolean value if the entries in this container is private.
+   * @return A boolean indicating if the entries in this container are private.
+   */
+  public boolean getIsPrivate() {
+    return isPrivate;
+  }
+
+  /**
+   * Gets the if of the {@link Account} that owns this container.
+   * @return The id of the parent {@link Account} of this container.
+   */
+  public short getParentAccountId() {
+    return parentAccountId;
+  }
+
+  /**
+   * Gets the metadata of the container.
+   * @return The metadata of the container.
+   * @throws JSONException If fails to compose metadata.
+   */
+  public JSONObject toJson() throws JSONException {
+    JSONObject metadata = new JSONObject();
+    metadata.put(CONTAINER_METADATA_VERSION_KEY, HIGHEST_CONTAINER_METADATA_VERSION);
+    metadata.put(CONTAINER_ID_KEY, id);
+    metadata.put(CONTAINER_NAME_KEY, name);
+    metadata.put(CONTAINER_STATUS_KEY, status);
+    metadata.put(CONTAINER_DESCRIPTION_KEY, description);
+    metadata.put(CONTAINER_IS_PRIVATE_KEY, isPrivate);
+    metadata.put(CONTAINER_PARENT_ACCOUNT_ID_KEY, parentAccountId);
+    return metadata;
+  }
+
+  /**
+   * Generates a String representation that uniquely identifies this container. The string
+   * is in the format of {@code Container[accountId:containerId]}.
+   * @return The String representation of this container.
+   */
+  @Override
+  public String toString() {
```

Comment:
can we have another method to print all details for a container. We might need it for trace logging. Same for account.